src/rvya25-profile.adoc

```diff
@@ -0,0 +1,466 @@
+only a minimal set of features and a large number of fine-grain
+extensions.
+
+Only user-mode (RVYA25U64) and supervisor-mode (RVYA25S64) profiles are
+specified in this family.
+
+NOTE: This profile is presented as incremental changes over RVA23.
+
+=== RVYA25U64 Profile
+
+The RVYA25U64 profile specifies the ISA features available to user-mode
+execution environments in 64-bit applications processors.  This is the
+most important profile within the application processor family in
+terms of the amount of software that targets this profile.
+
+==== RVYA25U64 Mandatory Base
+
+RVY64I is the mandatory base ISA for RVYA25U64 and is little-endian.  As
+per the unprivileged architecture specification, the `ECALL`
+instruction causes a requested trap to the execution environment.
+
+NOTE: The `ECALL` requirement is as specified in RVA23U64.
+
+==== RVYA25U64 Mandatory Extensions
+
+The following mandatory extensions are also mandatory in RVA23U64.
+
+- *M* Integer multiplication and division.
+- *A* Atomic instructions.
+- *F* Single-precision floating-point instructions.
+- *D* Double-precision floating-point instructions.
+- *B* Bit-manipulation instructions.
+- *Zicsr*  CSR instructions.  These are implied by presence of F.
+- *Zicntr* Base counters and timers.
+- *Zihpm* Hardware performance counters.
+- *Ziccif* Main memory regions with both the cacheability and
+  coherence PMAs must support instruction fetch, and any instruction
+  fetches of naturally aligned power-of-2 sizes up to min(ILEN,XLEN)
+  (i.e., 32 bits for RVYA25) are atomic.
+- *Ziccrse* Main memory regions with both the cacheability and coherence PMAs must support RsrvEventual.
+- *Ziccamoa* Main memory regions with both the cacheability and coherence PMAs must support all atomics in A.
+- *Zicclsm* Misaligned loads and stores to main memory regions with both the
+  cacheability and coherence PMAs must be supported.
+- *Za64rs* Reservation sets are contiguous, naturally aligned, and a
+   maximum of 64 bytes.
+- *Zihintpause* Pause hint.
+- *Zic64b* Cache blocks must be 64 bytes in size, naturally aligned in the
+address space.
+- *Zicbom* Cache-block management instructions.
+- *Zicbop* Cache-block prefetch instructions.
+- *Zicboz* Cache-Block Zero Instructions.
+- *Zfhmin* Half-precision floating-point.
+- *Zkt* Data-independent execution latency.
+- *V* Vector extension.
+- *Zvfhmin* Vector minimal half-precision floating-point.
+- *Zvbb* Vector basic bit-manipulation instructions.
+- *Zvkt* Vector data-independent execution latency.
+- *Zihintntl* Non-temporal locality hints.
+- *Zicond* Integer conditional operations.
+- *Zimop* may-be-operations.
+- *Zcmop* Compressed may-be-operations.
+- *Zcb* Additional compressed instructions.
+- *Zfa* Additional floating-Point instructions.
+- *Zawrs* Wait-on-reservation-set instructions.
+
+NOTE: The combination of "C" and "D" is incompatible with RVY as Zcd is an incompatible extension.
```

Comment:
Zcd can still be supported for Zyhybrid?

Reply:
Also, C should just mean "the subset of C that is valid for RVY". Making RV64YC be illegal is just going to be obnoxious to deal with.

Reply:
>Zcd can still be supported for Zyhybrid?

Of course, in that mode you get full RVA23 (I write that further down in the profile). It's the remapping of C.FLD->C.LY etc.

>Also, C should just mean "the subset of C that is valid for RVY". Making RV64YC be illegal is just going to be obnoxious to deal with.

ok - fair point - and this affects the spec as well as the profile - what you're saying is that C + D on RVY would _not_ include Zcd, you'd get C.LY[SP], C.SY[SP].

that works, but we then need to put an extension name back for C.LY[SP], C.SY[SP] so we can say that we get Zca + Zclsy (or whatever we call it). I agree it seems sensible to be able to specify C on the command line.

Reply:
https://github.com/riscv/riscv-cheri/pull/762

Reply:
done.